java-checks-test-sources/src/main/java/checks/NestedIfStatementsCheck.java

```diff
@@ -52,15 +54,15 @@ private void f() {
 
     for (int i = 0; i < 0; i++) { // Compliant - 1
       for (Object o: getObjects()) { // Compliant - 2
-        while (false) { // Compliant - 3
+        while (cond) { // Compliant - 3
 
-          for (int i = 0; i < 0; i++) { // Noncompliant
+          for (int j = 0; i < 0; i++) { // Noncompliant
```

Comment:
Why isn't local variable `j` used instead of `i`?

Reply:
Indeed, the code looks strange, but the code does not have to be meaningful as long as it compiles and test the rule correctly. I think it's fine to keep it.